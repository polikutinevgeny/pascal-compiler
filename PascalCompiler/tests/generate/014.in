begin
    write(4.0 > 3.0);
    write(3.0 > 4.0);

    write(4.0 < 3.0);
    write(3.0 < 4.0);

    write(4.0 >= 4.0);
    write(4.0 >= 3.0);
    write(3.0 >= 4.0);

    write(4.0 <= 4.0);
    write(4.0 <= 3.0);
    write(3.0 <= 4.0);

    write(2.0 = 2.0);
    write(2.0 <> 2.0);
    write(2.0 = 3.0);
    write(2.0 <> 3.0);
end.