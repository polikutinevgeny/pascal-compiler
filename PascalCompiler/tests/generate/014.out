.686p
include \masm32\include\masm32rt.inc
.mmx
.xmm
.code
main PROC
Push Ebp
Mov Ebp, Esp
Sub Esp, 0
Push 040100000h
Push 000000000h
Push 040080000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpnlesd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040080000h
Push 000000000h
Push 040100000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpnlesd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040100000h
Push 000000000h
Push 040080000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpltsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040080000h
Push 000000000h
Push 040100000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpltsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040100000h
Push 000000000h
Push 040100000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpnltsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040100000h
Push 000000000h
Push 040080000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpnltsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040080000h
Push 000000000h
Push 040100000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpnltsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040100000h
Push 000000000h
Push 040100000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmplesd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040100000h
Push 000000000h
Push 040080000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmplesd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040080000h
Push 000000000h
Push 040100000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmplesd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040000000h
Push 000000000h
Push 040000000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpeqsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040000000h
Push 000000000h
Push 040000000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpneqsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040000000h
Push 000000000h
Push 040080000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpeqsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 040000000h
Push 000000000h
Push 040080000h
Push 000000000h
Movsd Xmm1, qword ptr [Esp]
Add Esp, 8
Movsd Xmm0, qword ptr [Esp]
Add Esp, 8
Cmpneqsd Xmm0, Xmm1
Sub Esp, 4
Movd dword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Mov Esp, Ebp
Pop Ebp
exit
main ENDP
END main
